### ISSUE TITLE
EventTarget#addEventListener mishandles nullish "signal" option

### ISSUE THREAD

**Issue by Garcia101** [denoland/deno#14254]
Currently, Deno accepts this example snippet without error:
```js
window.addEventListener("load", null, {
 signal: null
});
```
According to <https://dom.spec.whatwg.org/#eventtarget>, EventTarget#addEventListener is defined as
```
interface EventTarget {
 ...
 undefined addEventListener(..., ..., optional (AddEventListenerOptions or boolean) options = {});
 ...
}

dictionary EventListenerOptions {
  boolean capture = false;
};

dictionary AddEventListenerOptions : EventListenerOptions {
  boolean passive = false;
  boolean once = false;
  AbortSignal signal;
};
```
Note the `AbortSignal signal`, which doesn't allow the `signal` value to be null, if the object property is present.

From a quick skim, it seems to be that this branch isn't being reached, even though it's intended to be?
https://github.com/denoland/deno/blob/8ae17026cbb30ab23fd79ac5cf5c207af5ad90a3/ext/web/02_event.js#L932-L934

**Comment by randomicon00** [comment]
From the spec, it says that the signal property can be null.

`signal (null or an [AbortSignal] object)`

**Comment by F3n67u** [comment]
> From the spec, it states that the signal property can be null.
> 
> `signal (null or an [AbortSignal] object)`

@randomicon00  I think null is not the javascript `null` value, it's more like undefined on spec text.


there is a wpt test which states a `TypeError` should throw when `signal` is `null`.

https://github.com/web-platform-tests/wpt/blob/3532a4b4082fbbb32438ae8670252e664aaa2efe/dom/events/AddEventListenerOptions-signal.any.js#L135-L138

**Comment by Garcia101** [comment]
> From the spec, it states that the signal property can be null.
> 
> `signal (null or an [AbortSignal] object)`

If anything, the code snippet that I provide yields different results than Deno when evaluated in a browser.
I expect the null is internal, and intended to mark handlers that aren't storing an AbortSignal reference.

**Comment by F3n67u** [comment]
@Garcia101 what's your deno version? "AddEventListenerOptions.signal shouldn't be nullable" fix is landed at version v1.12.0， so if you using version lower than version 1.12.0, then the problem will exist.

I am using version 1.20.6, the problem is not exist

```bash
$ cat test.js                                          
const et = new EventTarget();
et.addEventListener("foo", () => {}, { signal: null });

$ deno run test.js
error: Uncaught TypeError: signal must be non-null
et.addEventListener("foo", () => {}, { signal: null });
   ^
    at EventTarget.addEventListener (deno:ext/web/02_event.js:939:15)
    at file:///Users/feng/dev/deno/test.js:2:4
```

## Deno Version

```bash
$ deno --version   
deno 1.20.6 (release, x86_64-apple-darwin)
v8 10.0.139.6
typescript 4.6.2
```

**Comment by Garcia101** [comment]
> @Garcia101 what's your deno version? "AddEventListenerOptions.signal shouldn't be nullable" fix is landed at version v1.12.0， so if you using version lower than version 1.12.0, then the problem will exist.
> 
> fix pr: #11348 release note: https://github.com/denoland/deno/releases/tag/v1.12.0
> 
> I am using version 1.20.6, the problem is not exist
> 
> ```shell
> $ cat test.js                                          
> const et = new EventTarget();
> et.addEventListener("foo", () => {}, { signal: null });
> 
> $ deno run test.js
> error: Uncaught TypeError: signal must be non-null
> et.addEventListener("foo", () => {}, { signal: null });
>    ^
>     at EventTarget.addEventListener (deno:ext/web/02_event.js:939:15)
>     at file:///Users/feng/dev/deno/test.js:2:4
> ```
> 
> ## Deno Version
> 
> ```shell
> $ deno --version   
> deno 1.20.6 (release, x86_64-apple-darwin)
> v8 10.0.139.6
> typescript 4.6.2
> ```

I just ran this, to be fully certain:
```js
const event_target = new EventTarget();

event_target.addEventListener("abc", null, {
 signal: null
});

console.log(Deno.version);
```
output:
```
{ deno: "1.20.6+0e4574b", v8: "10.0.139.6", typescript: "4.6.2" }
```
I also tried on the non-canary release,
```
{ deno: "1.20.6", v8: "10.0.139.6", typescript: "4.6.2" }
```
Furthermore, note that I reference what happens to be the code intended to prevent this bug, in the issue message itself:
> From a quick skim, it seems to be that this branch isn't being reached, even though it's intended to be?
> https://github.com/denoland/deno/blob/8ae17026cbb30ab23fd79ac5cf5c207af5ad90a3/ext/web/02_event.js#L932-L934

That very code coincidentally comes from the PR that was intended to fix this issue.

The current implementation is erroneous and misaligned with the appropriate specification, as WebIDL type conversions are up-front, whereas this code is evaluated conditionally, that is, if the handler callback function is not `null`. Please do evaluate my suggested script snippet in your current version of Deno, and you will see the same. CrowlCats agreed that this is the case. So hopefully (Unfortunately, I don't have a means of locally building and testing Deno at the time, so my apologies for not attempting this myself.)

**Comment by F3n67u** [comment]
@Garcia101 my bad. I copy the snippet from the wpt test. I can reproduce this problem when I execute your snippet.
